docs/references/customization/appcharts.md

```diff
@@ -18,6 +18,108 @@ and
+  - A `Deployment`, `StatefulSet`, or similar main controller, specifying the `Pod` and
+    its containers running the application image.
+
+  - A `Service` to route web requests to the application.
+
+  - An [Ingress](lb.md) per specified route to front the `Service`.
+
+:::note
+
+Up to version 1 Epinio allowed only `Deployment` resources as the main controller of
+applications.
+
+:::
+
+:::tip
+
+Read the contents of the standard application chart.
+
+:::
+
+# Further Expectations
+
+Application Charts are expected to heed the following contraints to enable their use by
+Epinio:
+
+  1. It has to support all of the configuration values described in section
+     [Configuration](#configuration) below.
+
+  1. Pod resources have to have a number of annotations and labels for Epinio to locate
+     and use them, as described in section [Pods](#pods) below.
+
+:::note
+
+While a custom chart is free to support more configuration values Epinio will have no
+knowledge of them, and will not use them.
+
+:::
+
+## Configuration
+
+All configuration fields below are provied in the `epinio` hierarchy.
+
+|Name			|Kind			|Meaning							|
+|---			|---			|---								|
+|`appName`      	|string                 |The application's name						|
+|`configurations`	|sequence (string)      |The names of the configurations to import into the application	|
+|`env`          	|sequence (assignment)  |The application's environment variables and their values 	|
+|`imageURL`     	|string                 |A reference to the app image in Epinio's registry		|
+|`ingress`      	|string                 |The ingress class name to use, if any				|
+|`replicaCount` 	|integer                |The desired number of instances (pods) to deploy		|
+|`routes`       	|sequence (route)       |The routes (= domain+path+id) the app has to be reachable at	|
+|`stageID`      	|string                 |Id of the stage run which generated the app image		|
+|`start`        	|integer                |Optional. The time of deployment, in nanoseconds    		|
+|`tlsIssuer`    	|string                 |The name of the cert issuer to use for route certs		|
+|`username`     	|string                 |The name of the user deploying the application			|
+
+Routes are maps holding four keys
+
+|Name		|Kind	|Meaning								|
+|---		|---	|---									|
+|`domain`	|string	|The domain of the route						|
+|`id`		|string	|A unique id to name route `Ingress` and `Certificate` resources with	|
+|`path`		|string	|The sub-path of the route, if any					|
+|`secret`	|string	|Optional. Name of a `Secret` to directly use in securing the `Ingress`	|
+
+:::note
+
+The [Routing Secrets](routing_secrets.md) reference explains more about the `secrets` field.
+
+:::
+
+Environment assigment are maps holding two keys
+
+|Name	|Kind	|Meaning			|
+|---	|---	|---				|
+|`name`	|string	|The name of the variable	|
+|`value`|string	|The value of the variable	|
+
+## Pods
+
```

Comment:
```suggestion

Epinio adds the following annotations to the `Pods`:

```

Reply:
Note that the tables have the kind of added thing in the top-left corner. Still, no problem to be more explicit, also serving as separators